scenes/menu/main_menu.gd

```diff
@@ -17,14 +17,16 @@ const ModalDialog = preload("res://scenes/game/modal_dialog.gd")
 @onready var player_name_box : TextEdit = $PlayerNameBox
 @onready var replay_data_box : TextEdit = $EnterReplayBox
 
-@onready var start_ai_button : Button = $MenuList/VSAIBox/FightSettings/StartButton
-@onready var randomize_first_box : CheckBox = $MenuList/VSAIBox/FightSettings/RandomizeFirstCheckbox
+@onready var start_ai_button : Button = $MenuList/VSAIBox/StartButton
+@onready var randomize_first_box : CheckBox = $PreferencesWindow/VBoxContainer/AIFirstPlayerCheckbox
```

Comment:
I really want to avoid having one scene reference the internals of another scene like this.
Please don't do it this way.

What you probably want to do is have the PreferencesWindow change the GlobalSettings directly, then anything MainMenu cares about, it can just query the settings.
PreferencesWindow could also potentially emit a signal when it closes and this scene could listen to that signal and update any UI that it needs to update (or stop playing music, etc)

Reply:
The reason for this is, let's say we decide to change the layout of the preferences window so this checkbox isn't in the same parent structure.
That would then break this code, which is super annoying!